bindings/python/mainloop.pyx

```diff
@@ -0,0 +1,66 @@
+import asyncio
+
+
+cdef extern from "sol-mainloop.h":
+	struct sol_mainloop_implementation:
+		void run()
+		pass
+
+	bint sol_mainloop_set_implementation(const sol_mainloop_implementation *)
+
+
+cdef void wrap_sol_init():
+	pass
+
+cdef void wrap_sol_shutdown():
+	pass
+
+cdef void wrap_sol_quit():
+	asyncio.get_event_loop().stop()
+
+cdef void wrap_sol_run():
+	asyncio.get_event_loop().run_forever()
+
+cdef void *wrap_sol_timeout_add(args):
+	pass
+
+cdef void wrap_sol_timeout_del(args):
+	pass
+
+cdef void wrap_idle_add(args):
+	pass
+
+cdef void wrap_idle_del(args):
+	pass
+
+cdef void quit():
+	pass
+
+cdef void wrap_source_add(args):
+	pass
+
+cdef void wrap_source_del(args):
+	pass
+
+cdef void * wrap_source_get_data(args):
+	return NULL
+
+cdef sol_mainloop_implementation _py_asyncio_impl
+
+_py_asyncio_impl.run = wrap_sol_run
+
+
+"""_py_asyncio_impl.api_version = 1
```

Comment:
=1 will will cause you problems, you should use the define. I guess you can treat it as a regular C symbol, it will be a passtru.